lego/apps/surveys/tests/test_submissions_api.py

```diff
@@ -0,0 +1,97 @@
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from lego.apps.users.models import AbakusGroup, User
+
+
+def _get_list_url(survey_pk):
+    return reverse('api:v1:submission-list', kwargs={'survey_pk': survey_pk})
+
+
+def _get_detail_url(survey_pk, submission_pk):
+    return reverse('api:v1:submission-detail', kwargs={'survey_pk': survey_pk, 'pk': submission_pk})
+
+
+def submission_data(user):
+    return {
+        'user': user.id,
+        'survey': 1,
+        'answers': [],
+    }
+
+
+class SubmissionViewSetTestCase(APITestCase):
+    fixtures = ['test_users.yaml', 'test_abakus_groups.yaml', 'test_surveys.yaml',
+                'test_events.yaml', 'test_companies.yaml']
+
+    def setUp(self):
+        self.admin_user = User.objects.get(username='useradmin_test')
+        self.admin_group = AbakusGroup.objects.get(name='Bedkom')
+        self.admin_group.add_user(self.admin_user)
+        self.attended_user = User.objects.get(username='test1')
+        self.attending_group = AbakusGroup.objects.get(name='Abakus')
+        self.attending_group.add_user(self.attended_user)
+        self.regular_user = User.objects.get(username='abakule')
+
+    def test_create_admin(self):
+        """Admin users should be able to create submissions"""
+        self.client.force_authenticate(user=self.admin_user)
+        response = self.client.post(_get_list_url(1), submission_data(self.admin_user))
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_create_attended(self):
+        """Users who attended the event should not be able to create submissions"""
+        self.client.force_authenticate(user=self.regular_user)
+        response = self.client.post(_get_list_url(1), submission_data(self.attended_user))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_regular(self):
+        """Regular users should not be able to create submissions"""
+        self.client.force_authenticate(user=self.regular_user)
+        response = self.client.post(_get_list_url(1), submission_data(self.regular_user))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_detail_admin(self):
+        """Admin users should be able to see detailed submissions"""
```

Comment:
Should they? Are admins allowed to know what each attendee answered for each question and not just a summary/statistics?

Reply:
I was using Google Forms as a reference and they provide full access to the creators of forms, including individual answers, so that's what I did initially. Might want to display the submission but without username in the frontend, I guess.

Reply:
I think it's important that submissions are 'anonymous' for those reviewing them. It might be  a bit scary to give feedback on an event or Bedkom itself when you know that someone in Bedkom can read your name next to your comment.